### PR TITLE
[VL] Support uuid function

### PR DESCRIPTION
**Pull request by zhli1142015**
## What changes were proposed in this pull request?

Support uuid function.

## How was this patch tested?

UT.

### CODE REVIEW

**Comment by zhli1142015** [comment]
Run Gluten Clickhouse CI

**Comment by zhli1142015** [comment]
@PHILO-HE  and @rui-mo , could you help review this? thanks.

**Comment by zhli1142015** [comment]
Run Gluten Clickhouse CI

File: gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala [hunk 1]
```diff
@@ -199,6 +199,10 @@ trait SparkPlanExecApi {
     throw new UnsupportedOperationException("NaNvl is not supported")
   }
 
+  def genUuidTransformer(substraitExprName: String, original: Uuid): ExpressionTransformer = {
+    GenericExpressionTransformer(substraitExprName, Seq(), original)
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala:203]: Could you remind me why we distinguish the implementations of Velox and CH backend?

**Review by rui-mo — COMMENTED** [review]
Thanks. Rebase is needed.

File: gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala (hunk 1 above)
Comment (zhli1142015) [reply gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala:203]: Uuid is a leaf expression, looks CH just treat it as leaf expression and ignore the seed gerenated by spark.
Velox uuid can accept the seed parameter as a constant input. We get the seed and pass it to velox. This is the reason we distinguish the implements.
Thanks.

Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala:203]: cc: @zzcclp

**Review by PHILO-HE — COMMENTED** [review]
@zhli1142015, just one comment. Please also rebase the code. Thanks!

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala [hunk 2]
```diff
@@ -129,6 +129,13 @@ class SparkPlanExecApiImpl extends SparkPlanExecApi {
     )
   }
 
+  /** Transform Uuid to Substrait. */
+  override def genUuidTransformer(
+      substraitExprName: String,
+      original: Uuid): ExpressionTransformer = {
+    UuidTransformer(substraitExprName, original)
```
[the remark this answers is not in the document]
Comment (zhli1142015) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala]: Thanks, updated.

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Review by rui-mo — APPROVED** [review]

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**